### ISSUE TITLE
Fix checkpoint saving and loading

### ISSUE THREAD

**Issue by mworchel** [mworchel/svbrdf-estimation#31]
While the weights of the network and the current epoch are saved correctly, we do not store the optimizer (and scheduler) state right now. In order to continue training predictably, we should do so. This can serve as a reference https://pytorch.org/tutorials/beginner/saving_loading_models.html#saving-loading-a-general-checkpoint-for-inference-and-or-resuming-training. It also shows, how extra metadata like the epoch and current loss can be stored in the checkpoint file.
